# Clase 6

En esta clase revisaremos el material de 

[Unidad 2](./Chapter_3/SIC_AI_Ch03_Unit02.ipynb)

- Tratamos de ver la Unidad 3 pero no alcanzó el tiempo

- [Colab trabajados](./3.2_pandas/)

- Ejercicios de Fernanda - desde su repositorio [link](https://github.com/Fernanda-Mesa/Clase06_IA/blob/main/Clase06.ipynb)

- Ejercicios de Javier - desde su repositorio [link](https://github.com/XaxavierJS/sic_ai_2024/blob/main/subir.ipynb)

# Tarea sesión

Realizar un análisis de los datos con Pandas de alguna de los siguientes archivos y enviarlos a mi correo

- [Data Student List](./Chapter_3/data_studentlist.csv)

- [McDonalds Informacion](https://github.com/davidlealo/mujeresbigdatajun23coquimbo/blob/main/mcdonalds.xlsx)

In [13]:
import numpy as np
import pandas as pd

In [14]:
data=pd.read_csv('C:/Users/yael1/Documents/GitHub/sic_ai_2024/002_preprocesamiento/Chapter_3/data_studentlist.csv')

In [15]:
data

name gender  age  grade absence bloodtype  height  weight
0      Jared Diamond      M   23      3       Y         O   165.3    68.2
1     Sarah O'Donnel      F   22      2       N        AB   170.1    53.0
2       Brian Martin      M   24      4       N         B   175.0    80.1
3       David Hassel      M   23      3       N        AB   182.1    85.7
4    Clara Rodriquez      F   20      1       Y         A   168.0    49.5
5   Jennifer Lorentz      F   21      2       N         O   162.0    52.0
6        Susan Clark      F   22      1       N         O   155.2    45.3
7    Margareth Jones      F   23      1       N         A   176.9    55.0
8       John Bertsch      M   23      3       N         B   178.5    64.2
9     Jake Timmerman      M   22      2       N         B   176.1    61.3
10     Joshua Connor      M   24      4       Y         O   167.1    62.0
11      John Matsuda      M   22      2       N        AB   180.0    75.8
12      Eddy Johnson      M   21      1       N         A   162.2    55.3
13  Rebecah Anderson      F   23      3       N         O   176.1    53.1
14      Linda Carter      F   22      2       N         B   158.2    45.2
15    Richard Swayze      M   24      4       Y         B   168.6    70.2
16       Andrew King      M   21      1       N         A   169.2    62.2

In [14]:
data.describe

<bound method NDFrame.describe of                 name gender  age  grade absence bloodtype  height  weight
0      Jared Diamond      M   23      3       Y         O   165.3    68.2
1     Sarah O'Donnel      F   22      2       N        AB   170.1    53.0
2       Brian Martin      M   24      4       N         B   175.0    80.1
3       David Hassel      M   23      3       N        AB   182.1    85.7
4    Clara Rodriquez      F   20      1       Y         A   168.0    49.5
5   Jennifer Lorentz      F   21      2       N         O   162.0    52.0
6        Susan Clark      F   22      1       N         O   155.2    45.3
7    Margareth Jones      F   23      1       N         A   176.9    55.0
8       John Bertsch      M   23      3       N         B   178.5    64.2
9     Jake Timmerman      M   22      2       N         B   176.1    61.3
10     Joshua Connor      M   24      4       Y         O   167.1    62.0
11      John Matsuda      M   22      2       N        AB   180.0    75.8
12  

describe te da los estadísticos descriptivos básicos

In [15]:
data.describe()

age      grade      height     weight
count  17.000000  17.000000   17.000000  17.000000
mean   22.352941   2.294118  170.035294  61.064706
std     1.169464   1.104802    7.849756  11.851685
min    20.000000   1.000000  155.200000  45.200000
25%    22.000000   1.000000  165.300000  53.000000
50%    22.000000   2.000000  169.200000  61.300000
75%    23.000000   3.000000  176.100000  68.200000
max    24.000000   4.000000  182.100000  85.700000

info() te dice el tipo de dato

In [18]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17 entries, 0 to 16
Data columns (total 8 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   name       17 non-null     object 
 1   gender     17 non-null     object 
 2   age        17 non-null     int64  
 3   grade      17 non-null     int64  
 4   absence    17 non-null     object 
 5   bloodtype  17 non-null     object 
 6   height     17 non-null     float64
 7   weight     17 non-null     float64
dtypes: float64(2), int64(2), object(4)
memory usage: 1.2+ KB


type te ayuda a verificar si estás trabajando comn un dataframe o una serie u otro

In [16]:
type(data)

pandas.core.frame.DataFrame

In [19]:
data.head()
#te devuelve las primeras filas del df

name gender  age  grade absence bloodtype  height  weight
0    Jared Diamond      M   23      3       Y         O   165.3    68.2
1   Sarah O'Donnel      F   22      2       N        AB   170.1    53.0
2     Brian Martin      M   24      4       N         B   175.0    80.1
3     David Hassel      M   23      3       N        AB   182.1    85.7
4  Clara Rodriquez      F   20      1       Y         A   168.0    49.5

In [20]:
data.tail()
#entrega las últimos filas del df

name gender  age  grade absence bloodtype  height  weight
12      Eddy Johnson      M   21      1       N         A   162.2    55.3
13  Rebecah Anderson      F   23      3       N         O   176.1    53.1
14      Linda Carter      F   22      2       N         B   158.2    45.2
15    Richard Swayze      M   24      4       Y         B   168.6    70.2
16       Andrew King      M   21      1       N         A   169.2    62.2

columns te dice las columnas presentes del df

In [21]:
data.columns

Index(['name', 'gender', 'age', 'grade', 'absence', 'bloodtype', 'height',
       'weight'],
      dtype='object')

index te entrega la cantidad de filas con sus indices

In [22]:
data.index

RangeIndex(start=0, stop=17, step=1)

In [25]:
data.bloodtype

0      O
1     AB
2      B
3     AB
4      A
5      O
6      O
7      A
8      B
9      B
10     O
11    AB
12     A
13     O
14     B
15     B
16     A
Name: bloodtype, dtype: object

In [27]:
data[['grade', 'gender']]

grade gender
0       3      M
1       2      F
2       4      M
3       3      M
4       1      F
5       2      F
6       1      F
7       1      F
8       3      M
9       2      M
10      4      M
11      2      M
12      1      M
13      3      F
14      2      F
15      4      M
16      1      M

cada columna del dataframe corresponde a una serie

In [29]:
type(data["height"])

pandas.core.series.Series

unique() te permite saber los valores unicos posibles dentro de una columna de un dataframe, como poder ver las categor+ias de respuesta posibles de la encuesta

In [30]:

data["gender"].unique()

array(['M', 'F'], dtype=object)

In [32]:
data['bloodtype'].unique()

array(['O', 'AB', 'B', 'A'], dtype=object)

creo que lo principal es establecer que ha en total 17 estudiantes y que las edades de los y las estudiantes van desde ls 20 a los 24 años con una media de 22,35 y que las notas van de 1 a 4 con un promedio de 2,29. También que pesan entre 45,2 a 85,7 con una media de 61 y sus pesos oscilan desde 155,2 a los 182,1, se desconoce l medida con un promedio de 170.

In [34]:
data.describe()

age      grade      height     weight
count  17.000000  17.000000   17.000000  17.000000
mean   22.352941   2.294118  170.035294  61.064706
std     1.169464   1.104802    7.849756  11.851685
min    20.000000   1.000000  155.200000  45.200000
25%    22.000000   1.000000  165.300000  53.000000
50%    22.000000   2.000000  169.200000  61.300000
75%    23.000000   3.000000  176.100000  68.200000
max    24.000000   4.000000  182.100000  85.700000

astype permite cambiar el tipo de dato

In [36]:
data["weight"] = data["weight"].astype('int64')

In [37]:
data

name gender  age  grade absence bloodtype  height  weight
0      Jared Diamond      M   23      3       Y         O   165.3      68
1     Sarah O'Donnel      F   22      2       N        AB   170.1      53
2       Brian Martin      M   24      4       N         B   175.0      80
3       David Hassel      M   23      3       N        AB   182.1      85
4    Clara Rodriquez      F   20      1       Y         A   168.0      49
5   Jennifer Lorentz      F   21      2       N         O   162.0      52
6        Susan Clark      F   22      1       N         O   155.2      45
7    Margareth Jones      F   23      1       N         A   176.9      55
8       John Bertsch      M   23      3       N         B   178.5      64
9     Jake Timmerman      M   22      2       N         B   176.1      61
10     Joshua Connor      M   24      4       Y         O   167.1      62
11      John Matsuda      M   22      2       N        AB   180.0      75
12      Eddy Johnson      M   21      1       N         A   162.2      55
13  Rebecah Anderson      F   23      3       N         O   176.1      53
14      Linda Carter      F   22      2       N         B   158.2      45
15    Richard Swayze      M   24      4       Y         B   168.6      70
16       Andrew King      M   21      1       N         A   169.2      62

In [39]:
data.value_counts()

name              gender  age  grade  absence  bloodtype  height  weight
Andrew King       M       21   1      N        A          169.2   62        1
John Matsuda      M       22   2      N        AB         180.0   75        1
Sarah O'Donnel    F       22   2      N        AB         170.1   53        1
Richard Swayze    M       24   4      Y        B          168.6   70        1
Rebecah Anderson  F       23   3      N        O          176.1   53        1
Margareth Jones   F       23   1      N        A          176.9   55        1
Linda Carter      F       22   2      N        B          158.2   45        1
Joshua Connor     M       24   4      Y        O          167.1   62        1
John Bertsch      M       23   3      N        B          178.5   64        1
Brian Martin      M       24   4      N        B          175.0   80        1
Jennifer Lorentz  F       21   2      N        O          162.0   52        1
Jared Diamond     M       23   3      Y        O          165.3   68 

value_counts te permite saber la frecuencia

In [41]:
data['gender'].value_counts()

M    10
F     7
Name: gender, dtype: int64

In [40]:
data['bloodtype'].value_counts()

O     5
B     5
A     4
AB    3
Name: bloodtype, dtype: int64

In [43]:
data['absence'].value_counts()

N    13
Y     4
Name: absence, dtype: int64

In [44]:
data['age'].value_counts()

23    5
22    5
24    3
21    3
20    1
Name: age, dtype: int64

In [46]:
data['grade'][10]

4

el sort_values funciona solo con series

In [47]:
print(data.sort_values())

TypeError: sort_values() missing 1 required positional argument: 'by'

In [48]:
print(data['age'].sort_values())

4     20
16    21
5     21
12    21
1     22
14    22
6     22
11    22
9     22
13    23
0     23
7     23
3     23
8     23
15    24
2     24
10    24
Name: age, dtype: int64


In [49]:
print(data['grade'].sort_values())

16    1
4     1
6     1
7     1
12    1
1     2
14    2
5     2
9     2
11    2
13    3
0     3
3     3
8     3
15    4
2     4
10    4
Name: grade, dtype: int64


In [50]:
print(data['height'].sort_values())

6     155.2
14    158.2
5     162.0
12    162.2
0     165.3
10    167.1
4     168.0
15    168.6
16    169.2
1     170.1
2     175.0
9     176.1
13    176.1
7     176.9
8     178.5
11    180.0
3     182.1
Name: height, dtype: float64


In [17]:

data.head(15)

name gender  age  grade absence bloodtype  height  weight
0      Jared Diamond      M   23      3       Y         O   165.3    68.2
1     Sarah O'Donnel      F   22      2       N        AB   170.1    53.0
2       Brian Martin      M   24      4       N         B   175.0    80.1
3       David Hassel      M   23      3       N        AB   182.1    85.7
4    Clara Rodriquez      F   20      1       Y         A   168.0    49.5
5   Jennifer Lorentz      F   21      2       N         O   162.0    52.0
6        Susan Clark      F   22      1       N         O   155.2    45.3
7    Margareth Jones      F   23      1       N         A   176.9    55.0
8       John Bertsch      M   23      3       N         B   178.5    64.2
9     Jake Timmerman      M   22      2       N         B   176.1    61.3
10     Joshua Connor      M   24      4       Y         O   167.1    62.0
11      John Matsuda      M   22      2       N        AB   180.0    75.8
12      Eddy Johnson      M   21      1       N         A   162.2    55.3
13  Rebecah Anderson      F   23      3       N         O   176.1    53.1
14      Linda Carter      F   22      2       N         B   158.2    45.2

In [18]:
data.groupby(['gender','bloodtype'])['height'].mean()

gender  bloodtype
F       A            172.450000
        AB           170.100000
        B            158.200000
        O            164.433333
M       A            165.700000
        AB           181.050000
        B            174.550000
        O            166.200000
Name: height, dtype: float64

groupby sirve para agrupar según una columna o más categorías/columnas

In [19]:
data.groupby(['gender','bloodtype'])['weight'].mean()

gender  bloodtype
F       A            52.250000
        AB           53.000000
        B            45.200000
        O            50.133333
M       A            58.750000
        AB           80.750000
        B            68.950000
        O            65.100000
Name: weight, dtype: float64

In [20]:
data.groupby(['gender','bloodtype'])['age'].mean()

gender  bloodtype
F       A            21.50
        AB           22.00
        B            22.00
        O            22.00
M       A            21.00
        AB           22.50
        B            23.25
        O            23.50
Name: age, dtype: float64

In [21]:
data.groupby(['gender','bloodtype'])['grade'].mean()

gender  bloodtype
F       A            1.00
        AB           2.00
        B            2.00
        O            2.00
M       A            1.00
        AB           2.50
        B            3.25
        O            3.50
Name: grade, dtype: float64

In [22]:
data.groupby(['gender'])['grade'].mean()

gender
F    1.714286
M    2.700000
Name: grade, dtype: float64

los hombres o quienes se identifican con pronombre masculino tienen mejores notas

es hasta acá.

In [16]:

import matplotlib as mpl
import matplotlib.pyplot as plt

In [6]:
import seaborn as sns

In [4]:
from __future__ import annotations

In [5]:
%pip install seaborn

Note: you may need to restart the kernel to use updated packages.


You should consider upgrading via the 'c:\Users\yael1\AppData\Local\Programs\Python\Python38-32\python.exe -m pip install --upgrade pip' command.


In [10]:
df = sns.load_dataset('titanic') 

In [11]:

df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [26]:
my_dict = {"Size":["L", "L", "M", "M", "M", "S", "S", "S", "S"],
           "Type":["A", "A", "A", "B", "B", "A", "A", "B", "B"],
           "Location":["L1", "L1", "L1", "L2", "L2", "L1", "L2", "L2", "L1"],
           "A":[1, 2, 2, 3, 3, 4, 5, 6, 7],
           "B":[2, 4, 5, 5, 6, 6, 8, 9, 9]}
df = pd.DataFrame(my_dict)
df

Size Type Location  A  B
0    L    A       L1  1  2
1    L    A       L1  2  4
2    M    A       L1  2  5
3    M    B       L2  3  5
4    M    B       L2  3  6
5    S    A       L1  4  6
6    S    A       L2  5  8
7    S    B       L2  6  9
8    S    B       L1  7  9

In [27]:
dfr = pd.pivot_table(df, index=['Size','Type'], columns='Location', values='B')

In [28]:
dfr

Location    L1   L2
Size Type          
L    A     3.0  NaN
M    A     5.0  NaN
     B     NaN  5.5
S    A     6.0  8.0
     B     9.0  9.0